remote_version.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+
+	"github.com/hansmi/paperhooks/pkg/client"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type remoteVersionClient interface {
+	GetRemoteVersion(ctx context.Context) (*client.RemoteVersion, *client.Response, error)
+}
+
+type remoteVersionCollector struct {
+	cl remoteVersionClient
+
+	updateAvailableDesc *prometheus.Desc
+}
+
+func newRemoteVersionCollector(cl remoteVersionClient) *remoteVersionCollector {
+	return &remoteVersionCollector{
+		cl: cl,
+
+		updateAvailableDesc: prometheus.NewDesc("paperless_remote_version_update_available", "Whether an update is available.", []string{"version"}, nil),
+	}
+}
+
+func (c *remoteVersionCollector) describe(ch chan<- *prometheus.Desc) {
+	ch <- c.updateAvailableDesc
+}
+
+func (c *remoteVersionCollector) collect(ctx context.Context, ch chan<- prometheus.Metric) error {
+	remoteVersion, _, err := c.cl.GetRemoteVersion(ctx)
```

Comment:
Paperless-ngx [does not implement caching](https://github.com/paperless-ngx/paperless-ngx/blob/61cb5103ed96dbe6a4f44129f92616b4c0d99b45/src/documents/views.py#L2210-L2247) for the `remote_version` resource. As implemented every scrape issues an HTTP request to the GitHub API. Can we cache the result for a (configurable) amount of time, say 24 hours?

Paperless may not have internet access (mine doesn't). Every scrape would fail. I think the easiest option would be a boolean flag (maybe `allow-remote-network` so it'd be usable for other similar situations?) with its value being passed to `newCollector` where it can control scrapers.

Reply:
I added a new flag in https://github.com/hansmi/prometheus-paperless-exporter/pull/57/commits/a6f24cc2e781e12edc8313e92080edcda112fb8f to exclude the remote version collector.

About the caching piece, I am not sure if this needed. See https://github.com/paperless-ngx/paperless-ngx/pull/9229
I would not be in favor of adding caching into an monitoring exporter, as we aim to get the real view and not a cached one.

Reply:
Please invert the flag so that external network requests are disabled by default (`enable-remote-network`). Then I'm also fine without having caching. If someone enables that flag and has a very aggressive scrape interval then it's up to them.

Reply:
Done in https://github.com/hansmi/prometheus-paperless-exporter/pull/57/commits/6db6890b0fc9ee702423de8aa968fa652bebfaa6